src/pages/dashboard/staking-pools/SafetyPoolDetail.tsx

```diff
@@ -1,71 +1,174 @@
 
 import { DetailPageLayoutContainer, ContentLeft, ContentRight, CardRow } from '../DetailPageStyles';
 
+const FIFTEEN_MINUTES_SECONDS = 15 * 60;
+
 type SafetyPoolDetailProps = {} & LocalizationProps;
 
 const SafetyPoolDetail: React.FC<
   SafetyPoolDetailProps & ReturnType<typeof mapStateToProps> & ReturnType<typeof mapDispatchToProps>
 > = ({
+  addNotification,
   isUserGeoBlocked,
   openModal,
   stakingBalancesData,
   stakingPoolsData,
   stringGetter,
   walletAddress,
+  withdrawBalancesData,
 }) => {
+  const [isLoadingWithdrawAvailable, setIsLoadingWithdrawAvailable] = useState<boolean>(false);
+
   usePollStakingPoolsData();
   usePollStakingBalances();
 
-  const { balances } = stakingBalancesData;
+  usePollEpochData({ stakingPool: StakingPool.Safety });
+  usePollWithdrawBalances({ stakingPool: StakingPool.Safety });
+
+  const { userBalance, unclaimedRewards } = stakingBalancesData.balances[StakingPool.Safety];
+
+  const {
+    nextEpochDate,
+    lengthOfBlackoutWindow,
+    poolSize,
+    rewardsPerSecond,
+  } = stakingPoolsData.data[StakingPool.Safety];
 
-  const safetyPoolStakingBalances = balances[StakingPool.Safety];
-  const safetyPoolData = stakingPoolsData.data[StakingPool.Safety];
+  const formattedDiffUntilEpoch = useGetCountdownDiff({
+    futureDateISO: nextEpochDate,
+    stringGetter,
+  });
 
-  const userHasStakingBalance = walletAddress && safetyPoolStakingBalances.userBalance;
+  const blackoutWindowStartTime = DateTime.fromISO(nextEpochDate ?? '').minus({
+    seconds: Number(lengthOfBlackoutWindow ?? 0),
+  });
+
+  const { seconds: secondsUntilBlackoutWindow } = blackoutWindowStartTime.diff(
+    DateTime.local(),
+    'seconds'
+  );
 
-  let userBalance: React.ReactNode = '-';
+  const formattedDiffUntilBlackoutWindow = useGetCountdownDiff({
+    futureDateISO: blackoutWindowStartTime.toISO(),
+    stringGetter,
+  });
+
+  const { availableWithdrawBalance, pendingWithdrawBalance } = withdrawBalancesData[
+    StakingPool.Safety
+  ];
+
+  const userStakingBalance = calculateUserStakingBalance({
+    stakingBalancesData,
+    stakingPool: StakingPool.Safety,
+    withdrawBalancesData,
+  });
+
+  const userHasStakingBalance = walletAddress && userStakingBalance?.gt(0);
+
+  const handleOnClickWithdrawAvailable = async () => {
+    try {
+      setIsLoadingWithdrawAvailable(true);
+
+      const txHash = await contractClient.stakingPoolClient.withdrawAvailableBalance({
+        amount: availableWithdrawBalance,
+        stakingPool: StakingPool.Safety,
+        walletAddress: walletAddress as string,
+      });
+
+      addNotification({
+        notificationType: NotificationType.Withdraw,
+        notificationData: { txHash },
+      });
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setIsLoadingWithdrawAvailable(false);
+    }
+  };
+
+  let formattedUserBalance: React.ReactNode = '-';
+
+  if (walletAddress && userStakingBalance) {
```

Comment:
Multiple checks for `walletAddress`, I think we could wrap all of them and keep the new lines to visually separate.

Reply:
😬 fixed!